### PR TITLE
Feature/cc 2548

### PR DESCRIPTION
**Pull request by rDr4g0n**
* Create LogSearch angular service which makes it easy to generate a url to a specific kibana search
* Import rison encoder to convert js objects to the format kibana expects
* Make main nav "Log" link open Kibana in a new window with some sensible defaults
* Add "Service Logs" link to each service that opens Kibana in a new window, filtered down to that service
* Add "Instance Logs" link to each instance that opens Kibana in a new window, filtered down to that instance

### CODE REVIEW

**Comment by rDr4g0n** [comment]
retest this

File: web/ui/src/Services/ServiceDetailsController.js [hunk 1]
```diff
@@ -684,6 +684,40 @@
             }
         };
 
+        $scope.getServiceLogURL = function(service){
+            if(!service){
+                return "";
+            }
+            let a = LogSearch.getDefaultAppConfig(),
```
Comment (smousa) [inline web/ui/src/Services/ServiceDetailsController.js]: Can you make these variables more descriptive?
Also, add comments for what this is doing

File: web/ui/src/Services/ServiceDetailsController.js [hunk 2]
```diff
@@ -684,6 +684,40 @@
...
+                    query: `fields.service:${service.id} AND fields.instance:* AND message:*`
+                }
+            };
+            a.columns = ["fields.instance","message"];
+
+            return LogSearch.getURL(a, g);
+        };
+
+        $scope.getInstanceLogURL = function(instance){
+            let a = LogSearch.getDefaultAppConfig(),
```
Comment (smousa) [inline web/ui/src/Services/ServiceDetailsController.js]: See above

File: web/ui/src/common/LogSearch.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            return `
+${baseURL}/app/kibana?#/discover?
+_g=${rison.encode(globalConfig)}&
+_a=${rison.encode(appConfig)}&
+indexPattern=${index}&type=histogram`;
+    }
+
+    class LogSearch {
+        constructor(){
+            // TODO - use $location
```
Comment (smousa) [inline web/ui/src/common/LogSearch.js]: TODO? Is this still applicable?

**Pull request merged**